pkg/apis/workflow/v1alpha1/event_types.go

```diff
@@ -40,6 +40,17 @@ type Submit struct {
 	// WorkflowTemplateRef the workflow template to submit
 	WorkflowTemplateRef WorkflowTemplateRef `json:"workflowTemplateRef" protobuf:"bytes,1,opt,name=workflowTemplateRef"`
 
+	// WorkflowName optional means to customize the name of the submitted workflow
+	WorkflowName *WorkflowName `json:"workflowName,omitempty" protobuf:"bytes,2,opt,name=workflowName"`
+
 	// Arguments extracted from the event and then set as arguments to the workflow created.
-	Arguments *Arguments `json:"arguments,omitempty" protobuf:"bytes,2,opt,name=arguments"`
+	Arguments *Arguments `json:"arguments,omitempty" protobuf:"bytes,3,opt,name=arguments"`
+}
+
+type WorkflowName struct {
+	// Value is the literal value to use for the workflow name.
+	Value *AnyString `json:"value,omitempty" protobuf:"bytes,1,opt,name=value"`
+
+	// ValueFrom is the source for the workflow name
+	ValueFrom *ValueFrom `json:"valueFrom,omitempty" protobuf:"bytes,2,opt,name=valueFrom"`
```

Comment:
I don't think you need `value` or `valueFrom` at all

Reply:
What should I be using instead? I used those to stay somewhat similar to how parameters are defined.

Reply:
my comment is irrelevant - I want you to replace `workflowName` with `metadata`